Provide a Travis CI configuration for thrift4go.

matttproud opened the pull request:
This commit adds support for Travis CI (continuous integration) in
thrift4go such that commits against the master branch will trigger
an automatic build and test run against the "release" branch du jour
of Go.  All that needs to happen after this gets incorporated is that
thrift4go register itself with http://travis-ci.org/.

This would be immensely helpful in providing automatic verification of future changes.  :-)

matttproud committed:
Provide a Travis CI configuration for thrift4go.

This commit adds support for Travis CI (continuous integration) in
thrift4go such that commits against the master branch will trigger
an automatic build and test run against the "release" branch du jour
of Go.  All that needs to happen after this gets incorporated is that
thrift4go register itself with http://travis-ci.org/.

matttproud commented:
Hmm.  According to http://about.travis-ci.org/docs/user/getting-started/, that key must be set and to a valid configuration.  Since there is no pure C or Go implementation listed, I presume the Erlang one will have minimal overhead.  Is this OK?  I've had this successfully working with github.com/matttproud/golang_instrumentation for some time now.

pomack commented:
the lint checker for travis complains that the `otp_release` key is not specified

pomack commented:
The lint checker at least doesn't complain when I change `language` to `go`

pomack commented:
lint checker at http://lint.travis-ci.org/

matttproud committed:
Update conf. for modern Travis CI environment.

Travis CI's test environment changed grossly since my last commit to my
existing Go project:
- Debian distribution update from ${I_FORGOT_WHAT} to 11.10;
- Inclusion of some Go build tools made by who-knows-whom that exhibit a
  number of anomalies when compiling even the standard Go runtime
  libraries; and
- Changes in the the default test commands that are run depending on the
  value of the `language:` YAML key.

Key steps to remediation:
- Purging ANY go tools found on the system image; and
- Keeping the runtime language set to `erlang`, which despite lint
  warnings, continues to work and compile---unlike setting it to
  `go` or `ruby`, which triggers a rake-dependent workflow that is
  obviously incompatible with Go.

External evidence for anomalies:
- `language:go` and `language:ruby` to `language:erlang` fixes
   build and test workflow anomalies
   (before: https://github.com/matttproud/golang_instrumentation/commit/ac812287bcb84a6e72b1dd04f6da49f0b35fe3b6)
   and
   (after: https://github.com/matttproud/golang_instrumentation/commit/628071f2038d0a018e8d6642498b69ced0f19565).

matttproud commented:
So, if you want to take a look at this, I'd appreciate it:

1.) I am willfully ignoring the linter, since using erlang without a defined `otp_release` causes no problems as evidenced on `http://travis-ci.org/#!/matttproud/golang_instrumentation`.

2.) Using `go` or `ruby`, while it validates in the linter, actually produces anomalies when it comes time for building and testing.  Merely compare builds `28` and `29` from the previous link, as Travis CI throws an internal stack trace.

Is this OK?  It's what I'm using now for my own other Golang projects.

pomack commented:
Merged in. Not a fan, but I understand that the systems are inconsistent.

pomack commented:
Hey Matt, I've pushed it out to Travis-CI but it's failing to build because it can't find any tests http://travis-ci.org/#!/pomack/thrift4go/builds/1837313

matttproud commented:
OK.  I'll file another pull request to get this fixed.  Again, I am terribly sorry about this.  Iteratively testing (working) Travis CI is a pain in the rear.
